.gitlab-ci.yml

```diff
@@ -0,0 +1,48 @@
+---
+# Run all tests
+stages:
+  - lint
+  - deployment test
+
+before_script:
+  - ansible --version
+  - molecule --version
+
+variables:
+  GET_SOURCES_ATTEMPTS: "5"
+
+Ansible lint:
+  stage: lint
+  before_script:
+    - git clone https://github.com/lean-delivery/ansible-lint-rules.git ~/ansible-lint-rules
+  script:
+    - ansible-lint -c .ansible-lint `find . -regex ".*\.\(yml\)"`
```

Comment:
Please fix according to new rules. See ansible-development-kit as a sample.

Reply:
For the record: ansible-development-kit changes do not work for me and they won't until https://github.com/ansible/ansible-lint/issues/373 is fixed:

~/ansible-role-jenkins-slave$ ansible-lint . -c .ansible-lint
Examining /home/vagrant/ansible-role-jenkins-slave/tasks/main.yml of type tasks
Examining /home/vagrant/ansible-role-jenkins-slave/handlers/main.yml of type handlers
Examining /home/vagrant/ansible-role-jenkins-slave/meta/main.yml of type meta

So the only way at the moment is still to pass a list of playbooks to ansible-lint:

~/ansible-role-jenkins-slave$ ansible-lint -c .ansible-lint $(find . -type f -name \*.yml)
Examining ./requirements.yml of type playbook
Examining ./tasks/Linux/Linux.yml of type playbook
Examining /home/vagrant/ansible-role-jenkins-slave/tasks/Linux/selinux-support.yml of type playbook
Examining ./tasks/Linux/selinux-support.yml of type playbook
Examining ./handlers/main.yml of type playbook
Examining ./tasks/Win32NT/service.yml of type playbook
Examining ./vars/main.yml of type playbook
Examining ./vars/Windows.yml of type playbook
Examining ./defaults/main.yml of type playbook
Examining ./vars/default.yml of type playbook
Examining ./.gitlab-ci.yml of type playbook
Examining ./meta/main.yml of type playbook
Examining ./tasks/not-supported.yml of type playbook
Examining ./tasks/main.yml of type playbook
Examining ./tasks/Win32NT/Win32NT.yml of type playbook
Examining /home/vagrant/ansible-role-jenkins-slave/tasks/Win32NT/agent.yml of type playbook
Examining /home/vagrant/ansible-role-jenkins-slave/tasks/Win32NT/service.yml of type playbook
Examining ./vars/RedHat.yml of type playbook
Examining ./tasks/Win32NT/agent.yml of type playbook
Examining ./vars/Debian.yml of type playbook

Reply:
On the other hand, it makes no sence to pass playbooks to ansible-lint at all, it sees nothing:

```
~/ansible-role-jenkins-slave$ ansible-lint -c .ansible-lint .
Examining /home/vagrant/ansible-role-jenkins-slave/tasks/main.yml of type tasks
Examining /home/vagrant/ansible-role-jenkins-slave/handlers/main.yml of type handlers
Examining /home/vagrant/ansible-role-jenkins-slave/meta/main.yml of type meta
[405] Remote package tasks should have a retry
/home/vagrant/ansible-role-jenkins-slave/tasks/main.yml:9
Task/Handler: Install requirements
```

```
~/ansible-role-jenkins-slave$ ansible-lint -c .ansible-lint tasks/main.yml
Examining tasks/main.yml of type playbook
```

or 

```
~/ansible-role-jenkins-slave$ ansible-lint -c .ansible-lint /home/vagrant/ansible-role-jenkins-slave/tasks/main.yml
Examining /home/vagrant/ansible-role-jenkins-slave/tasks/main.yml of type playbook
```

Reply:
Obviously ansible-lint is not able to interpret complex instructions like:

```
- name: "Choose platform based task"
  include_tasks: "{{ platform }}"
  with_first_found:
    - "system/{{ ansible_os_family }}.yml"
    - "not-supported.yml"
  loop_control:
    loop_var: platform
```
but only simple ones like include/import_tasks..